### PR TITLE
Realtime search

### LINKED ISSUE #40: Suggestion: live filtering for the search dialog
Feels like it could be useful to filter the contents of the page "live" as users type their search term

### PR DESCRIPTION
**Pull request by goulinkh**
## Done

- Real-time search

## QA

- Go to https://specs-canonical-com-49.demos.haus/
- Search for a term
- Make sure that the cards that are visible contain the searched term

## Issue

Fixes #40 #43

### CODE REVIEW

**Comment by jnsgruk** [comment]
This is AWESOME. Thank you!

File: templates/_base.html [hunk 1]
```diff
@@ -165,6 +176,15 @@
         filterCards();
       }
 
+      function search(reset=false) {
+        if(reset) {
+          updateValueFilter("","query")
+        }else {
```
Comment (andesol) [inline templates/_base.html]:
```suggestion
        } else {
```

File: templates/_base.html [hunk 2]
```diff
@@ -199,6 +219,7 @@
         generateOptions(authorSelect, authorOptions);
         init();
       });
+      searchBoxInput.addEventListener("input", ()=>search());
```
Comment (andesol) [inline templates/_base.html]:
```suggestion
      searchBoxInput.addEventListener("input", () => search());
```

**Review by andesol — APPROVED** [review]
LGTM! Quite clever way of searching text in the cards. If this keeps growing I think we'll need to move the `js` parts into its own file...

(Just some formatting details)

Btw, I tried to rebuild the demo and I got 

```
received unexpected HTTP status: 504 Gateway Time-out
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```
Might be just a temporary thing, because I remember it worked on Friday

**Pull request merged**